gdal/frmts/vrt/vrtfilters.cpp

```diff
@@ -495,65 +489,69 @@ CPLErr VRTKernelFilteredSource::FilterData( int nXSize, int nYSize,
 /* -------------------------------------------------------------------- */
     if( eType == GDT_Float32 )
     {
+        float *pafSrcData = reinterpret_cast<float *>( pabySrcData );
+        float *pafDstData = reinterpret_cast<float *>( pabyDstData );
+
         int bHasNoData = FALSE;
         const float fNoData =
             static_cast<float>( m_poRasterBand->GetNoDataValue(&bHasNoData) );
 
-        for( int iY = 0; iY < nYSize; iY++ )
+        for( int nAxis = 0; m_bSeparable ? nAxis < 2 : nAxis < 1; ++nAxis)
```

Comment:
Perhaps introduce const int nAxisCount = m_bSeparable ? 2 : 1 and use it in the for condition